ocaml/xapi/xapi_host.ml

```diff
@@ -1339,16 +1339,18 @@ let apply_edition_internal  ~__context ~host ~edition ~additional =
   let params =
     ("current_edition", current_edition) ::
     license_server @ current_license_params @ additional in
+  let edition_str = edition in
+  (* rename edition variable to prevent namespace clashes with V6_interface *)
```

Comment:
Where then? I don't see a `let edition` or something like that in there?

Reply:
It could be renamed in the function definition itself: https://github.com/xapi-project/xen-api/pull/3391/files/634cfc05e57fb3fbaab6312554d0c34dab103857..733531584953d17ce23901200b4f06e6db122086#diff-126a0619d5403393ed49475b5dd96aa4R1326

Reply:
It can, but why?

Reply:
You are right, I thought we were in the same situation as in `v6d` but here it's not needed

Reply:
As I understand it, prior to locally opening `V6_interface`, `edition` is interpreted as a string. Within the V6_interface namespace, it is interpreted as an RPC type instead of a string, which upsets the compiler.

Which do you think is the best solution?
1. Rename the `edition` type in the V6 interface to avoid clashes
2. Rename to `edition_str` within the function definition (this requires updating elsewhere as well)
3. Local rename for this small snippet of code (used in this PR)

Reply:
Or 4: don't locally open `V6_interface` anymore, but just use `V6_interface.`.

Reply:
Also works!